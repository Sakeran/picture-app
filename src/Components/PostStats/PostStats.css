.PostStats {
  max-width: 75%;
  min-width: 300px;
  margin: 0 auto;
  padding: 1em;
  background: white;
  display: flex;
  justify-content: center;
  align-items: center;
}

.PostStats button {
  color: #380331;
  font-size: 0.9em;
  margin-right: 1em;
  text-align: center;
  background-color: #C7FCCE;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.PostStats button i {
  margin-right: 0.25em;
}

.PostStats button:hover,
.PostStats button:active {
  margin-right: 2em;
  transform: scale(1.1);
  transition: transform 0.2s ease-in-out, margin-right 0.2s ease-in-out;
  background-color: #C7FCCE;
  color: #380331;
  cursor: pointer;
}

.PostStats-is-liked {
  background-color: #43d056;
  color: #cf0c0c;
  font-size: 1.1em;
}

.PostStats-stats {
  font-weight: lighter;
}
