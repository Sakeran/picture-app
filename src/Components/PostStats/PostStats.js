import React from 'react';
import PropTypes from 'prop-types';

import './PostStats.css';

const getButtonClass = ({liked}) => ('fa fa-2x ' + (
  liked ?
  'fa-heart PostStats-is-liked'
  :
  'fa-heart-o'
));

const PostStats = (props) => (
  <div className="PostStats section-border">
    <div className="PostStats-btn-container">
      <button className="button" onClick={props.likeFunc}>
        <i className={getButtonClass(props)} aria-hidden="true"></i>
        <div>
          {props.liked ? "Unlike" : "Like"}
        </div>
      </button>
    </div>
      {props.likeCount || '0' } people like this post.
  </div>
);

PostStats.propTypes = {
  likeCount: PropTypes.number,
  likeFunc: PropTypes.func
};

export default PostStats;
