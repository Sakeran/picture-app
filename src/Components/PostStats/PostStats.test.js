import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostStats from './PostStats';

test('renders correctly when "liked" is false', () => {
  const component = shallow(<PostStats liked={false} likeCount={0} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('renders correctly when "liked" is true', () => {
  const component = shallow(<PostStats liked={true} likeCount={1} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
