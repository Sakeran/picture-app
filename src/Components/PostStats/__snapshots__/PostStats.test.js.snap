// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when "liked" is false 1`] = `
<div
  className="PostStats"
>
  0 people like this post.
</div>
`;

exports[`renders correctly when "liked" is true 1`] = `
<div
  className="PostStats"
>
  1 person likes this post.
</div>
`;
